### PR TITLE
eg pylint fixes

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_messaging_shared.py [hunk 1]
```diff
@@ -13,18 +13,22 @@
...
 
-def _get_json_content(obj):
+def _get_json_content(obj): # pylint: disable=inconsistent-return-statements
     """Event mixin to have methods that are common to different Event types
     like CloudEvent, EventGridEvent etc.
+
+    :param obj: The object to get the JSON content from.
+    :type obj: Any
+    :rtype: Any
+    :return: The JSON content from the object.
```
Comment (kashifkhan) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_messaging_shared.py:24]: would this fit here ?
```suggestion
    :return: The JSON content from the object.
    :raises ValueError: If the provided JSON is invalid.
```

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py [hunk 2]
```diff
@@ -38,6 +37,7 @@ def generate_sas(endpoint, shared_access_key, expiration_date_utc, **kwargs):
     :keyword str api_version: The API Version to include in the signature.
      If not provided, the default API version will be used.
     :rtype: str
+    :return: A shared access signature string.
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py:40]: same as Kashif pointed out here :) 
https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py [hunk 3]
```diff
@@ -141,9 +141,14 @@ def _cloud_event_to_generated(cloud_event, **kwargs):
         **kwargs
     )
 
-def _from_cncf_events(event):
+def _from_cncf_events(event): # pylint: disable=inconsistent-return-statements
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py:144]: just curious but, is this generated code? if not, is it not annotated b/c it's internal?

Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py:144]: not generated code!

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py [hunk 4]
```diff
@@ -141,9 +142,14 @@ def _cloud_event_to_generated(cloud_event, **kwargs):
...
     """This takes in a CNCF cloudevent and returns a dictionary.
     If cloud events library is not installed, the event is returned back.
+
+    :param event: The event to be serialized
+    :type event: cloudevents.CloudEvent
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py:150]: Not sure if this is referring to the CloudEvents in core or if there is one defined in EG (and I don't see a `cloudevents` imported in this file) but I believe it should look more like this:
```suggestion
    :type event: ~azure.core.messaging.CloudEvent
```

**Review by swathipil — APPROVED** [review]
one comment but o/w lgtm!

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py [hunk 5]
```diff
@@ -106,6 +106,7 @@ def from_json(cls, event):
          a storage QueueMessage, eventhub's EventData or ServiceBusMessage
         :type event: object
         :rtype: EventGridEvent
+        :returns: An EventGridEvent object.
```
Comment (kashifkhan) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py:109]:
```suggestion
        :return: An EventGridEvent object.
```

**Review by kashifkhan — APPROVED** [review]

**Comment by kashifkhan** [comment]
Will this PR affect this issue #30319

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py (hunk 4 above)
Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py:150]: This is referring to a CNCF cloudEvent from the package cloudevents but it should be cloudevents.http.CloudEvent

**Pull request closed without merging**